client/client.go

```diff
@@ -288,18 +292,29 @@ func (c *client) GetAllMembers(ctx context.Context) ([]*pdpb.Member, error) {
 	start := time.Now()
 	defer func() { cmdDurationGetAllMembers.Observe(time.Since(start).Seconds()) }()
 
-	ctx, cancel := context.WithTimeout(ctx, c.timeout)
-	resp, err := c.leaderClient().GetMembers(ctx, &pdpb.GetMembersRequest{
-		Header: c.requestHeader(),
-	})
+	cctx, cancel := context.WithTimeout(ctx, c.timeout)
+	req := &pdpb.GetMembersRequest{Header: c.requestHeader(c.GetLeaderAddr())}
+	resp, err := c.leaderClient().GetMembers(cctx, req)
 	cancel()
+
 	if err != nil {
+		if rpcErr, ok := status.FromError(err); ok && isNetworkError(rpcErr.Code()) {
+			followerClient := c.followerClient()
+			if followerClient != nil {
+				cctx, cancel := context.WithTimeout(ctx, c.timeout)
+				resp, err = followerClient.GetMembers(cctx, req)
+				cancel()
+				if err == nil {
+					return resp.GetMembers(), nil
+				}
+			}
+		}
+
 		cmdFailDurationGetAllMembers.Observe(time.Since(start).Seconds())
```

Comment:
I think we should add a label to distinguish whether this is a `redirect` or `direct` request.

Reply:
I have changed the way to send the unary request, PTAL